tests/test-rest-nav-menu-items-controller.php

```diff
@@ -23,85 +45,760 @@ public static function wpSetUpBeforeClass( $factory ) {
+				'menu-item-object'    => 'post_tag',
+				'menu-item-object-id' => $this->tag_id,
+				'menu-item-status'    => 'publish',
+			)
+		);
 	}
 
+	/**
+	 *
+	 */
 	public function test_register_routes() {
+		$routes = rest_get_server()->get_routes();
+
+		$this->assertArrayHasKey( '/wp/v2/menu-items', $routes );
+		$this->assertCount( 2, $routes['/wp/v2/menu-items'] );
+		$this->assertArrayHasKey( '/wp/v2/menu-items/(?P<id>[\d]+)', $routes );
+		$this->assertCount( 3, $routes['/wp/v2/menu-items/(?P<id>[\d]+)'] );
 	}
 
+	/**
+	 *
+	 */
 	public function test_context_param() {
+		// Collection
+		$request  = new WP_REST_Request( 'OPTIONS', '/wp/v2/menu-items' );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+		$this->assertEquals( 'view', $data['endpoints'][0]['args']['context']['default'] );
+		$this->assertEquals( array( 'view', 'embed', 'edit' ), $data['endpoints'][0]['args']['context']['enum'] );
+		// Single
+		$request  = new WP_REST_Request( 'OPTIONS', '/wp/v2/menu-items/' . $this->menu_item_id );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+		$this->assertEquals( 'view', $data['endpoints'][0]['args']['context']['default'] );
+		$this->assertEquals( array( 'view', 'embed', 'edit' ), $data['endpoints'][0]['args']['context']['enum'] );
+	}
+
+	/**
+	 *
+	 */
+	public function test_registered_query_params() {
+		$request    = new WP_REST_Request( 'OPTIONS', '/wp/v2/menu-items' );
+		$response   = rest_get_server()->dispatch( $request );
+		$data       = $response->get_data();
+		$properties = $data['endpoints'][0]['args'];
+		$this->assertArrayHasKey( 'before', $properties );
+		$this->assertArrayHasKey( 'context', $properties );
+		$this->assertArrayHasKey( 'exclude', $properties );
+		$this->assertArrayHasKey( 'include', $properties );
+		$this->assertArrayHasKey( 'menu_order', $properties );
+		$this->assertArrayHasKey( 'menus', $properties );
+		$this->assertArrayHasKey( 'menus_exclude', $properties );
+		$this->assertArrayHasKey( 'offset', $properties );
+		$this->assertArrayHasKey( 'order', $properties );
+		$this->assertArrayHasKey( 'orderby', $properties );
+		$this->assertArrayHasKey( 'page', $properties );
+		$this->assertArrayHasKey( 'per_page', $properties );
+		$this->assertArrayHasKey( 'search', $properties );
+		$this->assertArrayHasKey( 'slug', $properties );
+		$this->assertArrayHasKey( 'status', $properties );
 	}
 
+	/**
+	 *
+	 */
+	public function test_registered_get_item_params() {
+		$request  = new WP_REST_Request( 'OPTIONS', sprintf( '/wp/v2/menu-items/%d', $this->menu_item_id ) );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+		$keys     = array_keys( $data['endpoints'][0]['args'] );
+		sort( $keys );
+		$this->assertEquals( array( 'context', 'id' ), $keys );
+	}
+
+	/**
+	 *
+	 */
 	public function test_get_items() {
+		wp_set_current_user( self::$admin_id );
```

Comment:
Do we have a test that ensures this errors out when the current user is logged out, or has insufficient permissions?

Reply:
These were added in https://github.com/WP-API/menus-endpoints/pull/40